### ISSUE TITLE
Provide access to item attributes

### ISSUE THREAD

**Issue by celinval** [rust-lang/project-stable-mir#83]
A definition in Rust can have attributes attached to them, .e.g., tool attributes. They are used by different tools to add metadata to the user code that can provide more information on how the code must be analyzed.

StableMIR should provide an API to retrieve the attributes of a crate definition. We can probably add a new method to the `CrateDef` that returns a list of `Attributes` attached to them.

**Comment by adwinwhite** [comment]
`rustc_ast::ast::Attribute`'s spec is quite complicated so we probably don't want to mirror it.


My plan:

- define stable `Attribute` similar to `Span`, merely as an index to internal struct. 
- add two most needed methods `value_str` and `path` for our stable `Attribute`.

Is this design okay?

**Comment by celinval** [comment]
That's a great start! Maybe add `span()` too.

I would like to eventually expose the `tokens()` method which somehow returns the `proc_macro::TokenStream`, so users can use crates such as `syn` to extract the contents of the attribute. That said, I haven't done any thorough investigation to see if that would be feasible and a reasonable implementation.

**Comment by celinval** [comment]
Or we could provide a `to_string()` for now (https://github.com/rust-lang/rust/blob/c290e9de32e8ba6a673ef125fde40eadd395d170/compiler/rustc_ast_pretty/src/pprust/mod.rs#L75)

**Comment by adwinwhite** [comment]
Good idea. Perhaps I should start with `span()` and `to_string()` for now. `String` can be easily parsed into `TokenStream` if needed. 

`tokens()` could save us some parsing, but it would also introduce `rustc_ast` as a dependency to `stable_mir` which has basically no dependency for now.

**Comment by celinval** [comment]
Let's resolve this for now. We can always add `token()` later. Thanks again @adwinwhite
